controllers/sensor/signal-filter.go

```diff
@@ -25,38 +25,121 @@ import (
+	se.log.Info().Interface("policy", policy).Msg("printing policy")
+	event := common.GetK8Event(&common.K8Event{
+		Name: policy.Name,
+		Namespace: se.sensor.Namespace,
+		ReportingInstance: se.sensor.Name,
+		ReportingController: se.sensor.Name,
+		Labels: map[string]string{
+			common.LabelEventSeen: "",
+			common.LabelSignalName: signalFilterName,
+		},
+		Type: common.LabelArgoEventsEscalationKind,
+		Action: string(policy.Level),
+		Reason: policy.Message,
+	})
+	err := common.CreateK8Event(event, se.kubeClient)
+	return err
+}
+
 // apply the signal filters to an event
-func filterEvent(f v1alpha1.SignalFilter, event *v1alpha.Event) (bool, error) {
-	dataRes, err := filterData(f.Data, event)
-	return filterTime(f.Time, &event.Context.EventTime) && filterContext(f.Context, &event.Context) && dataRes, err
+func (se *sensorExecutor) filterEvent(f v1alpha1.SignalFilter, event *v1alpha.Event) (bool, error) {
+	dataRes, err := se.filterData(f.Data.Filters, event)
+	// generate sensor failure event and mark sensor as failed
+	if err != nil {
+		return false, err
+	}
+	if !dataRes {
+		err = se.createEscalationEvent(f.Data.EscalationPolicy, f.Name)
+		if err != nil {
+			return false, err
+		}
+	}
+	timeRes, err := se.filterTime(f.Time, &event.Context.EventTime)
+	// generate sensor failure event and mark sensor as failed
+	if err != nil {
+		return false, err
+	}
+	if !timeRes {
+		err = se.createEscalationEvent(f.Time.EscalationPolicy, f.Name)
+		if err != nil {
+			return false, err
+		}
+	}
+	ctxRes := se.filterContext(f.Context, &event.Context)
+	if !ctxRes {
+		err = se.createEscalationEvent(f.Context.EscalationPolicy, f.Name)
+		if err != nil {
+			return false, err
+		}
+	}
+	return timeRes && ctxRes && dataRes, err
 }
 
 // applyTimeFilter checks the eventTime against the timeFilter:
 // 1. the eventTime is greater than or equal to the start time
 // 2. the eventTime is less than the end time
 // returns true if 1 and 2 are true and false otherwise
-func filterTime(timeFilter *v1alpha1.TimeFilter, eventTime *metav1.Time) bool {
-	if timeFilter != nil && eventTime != nil {
-		if timeFilter.Start != nil && timeFilter.Stop != nil {
-			return (timeFilter.Start.Before(eventTime) || timeFilter.Start.Equal(eventTime)) && eventTime.Before(timeFilter.Stop)
+func (se *sensorExecutor) filterTime(timeFilter *v1alpha1.TimeFilter, eventTime *metav1.Time) (bool, error) {
+	if timeFilter != nil {
+		currentT := time.Now().UTC()
+		se.log.Info().Str("current-time", currentT.String()).Msg("current time")
+		currentTStr := fmt.Sprintf("%d-%s-%d", currentT.Year(), int(currentT.Month()), currentT.Day())
+
+		if timeFilter.Start != "" && timeFilter.Stop != "" {
+			se.log.Info().Str("start time format", currentTStr + " " + timeFilter.Start).Msg("start time format")
+			startTime, err := time.Parse("2006-01-02 15:04:05", currentTStr + " " + timeFilter.Start)
```

Comment:
let's make the format string `2006-01-02 15:04:05` a constant since we use it more than once and seeing this may be confusing for some.

Reply:
done :+1: